### ISSUE TITLE
C# parser does not parse base type list of enums well if there are additional tokens

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#81117]
**Version Used**: VS 18.0.0 Insiders

**Steps to Reproduce**:
Paste this code:
```csharp
enum E : int, int
{
    None = 0
}
```

**Expected Behavior**:
The parser handles this scenario better:
* The base type list is parsed correctly like it is for other types.
* Error diagnostic are emitted as necessary, stating that enums can't have multiple base types.
* The opening brace token is parsed as the open brace token of the enum and normal parsing resumes from there.

**Actual Behavior**:
Any additional non-whitespace token between the first base type and the opening brace will disrupt the parser and lead to a strange syntax tree:

<img width="1216" height="592" alt="Image" src="https://github.com/user-attachments/assets/53423a2f-48d7-4642-8094-5f3fccaea26f" />

Multiple diagnostics are reported, none of which aid in understanding the error:

<img width="459" height="201" alt="Image" src="https://github.com/user-attachments/assets/620b584c-14a7-498f-a26f-f27094700171" />

By contrast, the VB parser is more resilient to this kind of situation:

<img width="442" height="128" alt="Image" src="https://github.com/user-attachments/assets/c603bda9-7f11-4597-8f02-fef3b2bf2d44" />

**Comment by CyrusNajmabadi** [comment]
This doesnt seen like a realistic scenario.

**Comment by CyrusNajmabadi** [comment]
Going to close it. Between all tokens/nodes, there are huge number if things that can be written that will through the parser off. We address things that are realistic states users commonly run into.  This doesn't apply here.
